packages/nextjs/src/utils/instrumentServer.ts

```diff
@@ -245,8 +264,11 @@ function makeWrappedReqHandler(origReqHandler: ReqHandler): WrappedReqHandler {
               metadata: { requestPath: reqPath },
               ...traceparentData,
             },
-            // extra context passed to the `tracesSampler`
-            { request: req },
+            // Extra context passed to the `tracesSampler` (Note: this weird format is in order to not break people's
+            // `tracesSampler` functions, even though the format of `nextReq` has changed (see note above re: nextjs
+            // 12.0.9). If `nextReq === req` (the old format), then we'll just spread the same stuff twice. If `nextReq`
+            // contains `req` (the new format), then this mimics the old format by flattening the data.)
```

Comment:
nit: I'd avoid using "old" and "new" words and use the versions directly. I find it less confusing, more to the point, and easier to build on top of in case we need to expand to different formats. This is a thing  you already do in the note above.

I suggest rewriting it to something like:
```suggestion
            // Extra context passed to the `tracesSampler`.
            // Next v12.0.9 incorporates a change to `nextReq` (see note above), and requires the following change to not crash:
            // In <12.0.9, `nextReq === req` and we spread the same stuff twice.
            // In >=12.0.9, `nextReq` contains `req` and we mimic the old format by flattening the data.
```

On the other hand, if the difference is whether `nextReq === req` is `true`, what is stopping us from making that comparison and returning the appropriate object instead of overwritting? I also find it easier to read.

Reply:
Fair points. I've reworded it.